src/Core/Content/Product/SearchKeyword/ProductSearchTermInterpreter.php

```diff
@@ -18,8 +18,15 @@
 #[Package('buyers-experience')]
 class ProductSearchTermInterpreter implements ProductSearchTermInterpreterInterface
 {
+    private const DEFAULT_MAX_CHARACTER_COUNT = 60;
+
     private const RELEVANT_KEYWORD_COUNT = 8;
 
+    /**
+     * @var array<string, int>
+     */
+    private array $maxCharacterCount = [];
```

Comment:
You need to add a [ResetInterface](https://developer.shopware.com/docs/v6.4/resources/references/adr/cache/2022-03-09-reset-class-state-during-requests.html) for the `maxCharacterCount`  to this class, as it shares it's instance in other classes like `ProductSearchBuilder`

Reply:
Ah, I wasn't aware of that. I've decided to outsource the whole character count limitation into a new service `CharacterLimiter` to keep the `ProductSearchTermInterpreter` class clean - the same way it has been done with the `TokenFilter`. The `CharacterLimiter` service also implements the `ResetInterface`.